client/src/modules/blockEditor/actions.ts

```diff
@@ -114,11 +167,48 @@ export const removeExportOfSelection = () => {
         return;
       }
 
-      const change = block.value
-        .change()
-        .unwrapInlineByKey(matchingNodes[0].key, {
-          data: { pointerId: hoveredItem.id }
-        });
+      const nodeToRemove = matchingNodes[0];
+
+      const ancestorNodes = block.value.document.getAncestors(nodeToRemove.key);
+
+      const isNested = ancestorNodes.find(ancestor => ancestor.type === "pointerExport");
+
+      let change;
+      change = block.value.change();
+
+      // delete spacer at start and end
```

Comment:
The spacer deletion here and below feels brittle. Can this be handled using the normalization process somehow?

Reply:
This code checks to see whether the spacer character exists at the start and end of the text of the pointer to be removed, and before and after the pointer to be removed. If the spacer character exists at any of those positions, it's deleted. 

If any of those nodes don't exist, or if the character at the relevant position isn't the spacer character, then the deletion doesn't occur. So this code might be brittle in some ways, but it abstracts over the spacer character, and it's cautious about when it actually deletes a character. I'm not too worried about it.

That said, I'd be happy to find a more robust way of doing this.

One thing we could do is expand the normalization process to make sure that *both* **(1)** the spacer character (currently thin space) occurs at the start/end of, and before/after, every pointer and **(2)** the spacer character *only* occurs in those positions (so delete all of the spacer characters not in those positions). 

Currently, we only do **(1)**, and allow users to insert additional spacer characters in their editors if they wish to. Pursuing **(2)** would take all of this "pointer removal cleanup" and include it in the normalization process, but it would mean users can't use thin spaces at their discretion (which might be fine).